subsys/nrf_security/Kconfig.legacy

```diff
@@ -22,7 +22,7 @@ config MBEDTLS_MEMORY_BUFFER_ALLOC_C
 
 config MBEDTLS_THREADING_C
 	bool "Threading support for Mbed TLS and PSA crypto"
-	default y if CC3XX_BACKEND || MBEDTLS_PSA_CRYPTO_C
+	default y if CC3XX_BACKEND || (MBEDTLS_PSA_CRYPTO_C && MULTITHREADING)
```

Comment:
Why not instead have a `depends on MULTITHREADING`?
This change here doesn't cover `CC3xx_BACKEND`; is it intended (and why)?

Reply:
The reasoning that I didn't do that before is that the CC3XX has also some atomic mutexes that can use internally. So technically it doesn't require Zephyr mutexes always. But I see now that even when these atomic mutexes are used we still require the application to enable the multithreading configuration. So I can try to add that as a dependency I think.